library/core/src/bool.rs

```diff
@@ -31,4 +31,17 @@ impl bool {
     pub fn then<T, F: FnOnce() -> T>(self, f: F) -> Option<T> {
         if self { Some(f()) } else { None }
     }
+
+    /// Returns the opposite value.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// assert_eq!(false.not(), true);
+    /// assert_eq!(true.not(), false);
+    /// ```
+    #[unstable(feature = "bool_not_method", issue = "none")]
+    pub fn not(self) -> Self {
```

Comment:
This method could be a `const fn`, did you decide not to make it one to keep it identical to the `Not::not` method?

Reply:
Tangentially, it being `const`able already can also be an added argument for making it an inherent method

Reply:
Note that the intention is to make it possible to make trait impls const also, but we're pretty far from seeing that stabilized.